### PR TITLE
Add Signer

### PR DESCRIPTION
**Pull request by woomurf**
- Add signer class to inject sign function from outside.
- Replace `signTransaction` and `getImpliedAddress` to signer's function.

### CODE REVIEW

File: src/signer.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import Wallet from "./wallet";
+
+export interface Signer {
+    getAddress(address?: string): string;
+    signMessage(message: any, address?: string): Promise<string> | string;
+}
+
+export class DefaultSigner implements Signer {
```
Comment (platfowner) [inline src/signer.ts:8]: Could you add some js doc comments?

**Review by platfowner — APPROVED** [review]
LGTM with some comments.
Thanks!

File: src/signer.ts (hunk 1 above)
Comment (woomurf) [reply src/signer.ts:26]: Added. Thanks!!

**Comment by woomurf** [comment]
@shyun-comcom @minjaeli @YunsuBae

**Comment by woomurf** [comment]
Thanks for the review!

**Pull request merged**